internal/github/user.go

```diff
@@ -0,0 +1,63 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/google/go-github/v47/github"
+)
+
+type Owners struct {
```

Comment:
This type could have a `AllUsernames` method that would return a set of all usernames in the file. The we could have fast lookups.

Reply:
Not sure I understand your suggestion.
1. Why is it a "faster lookup" than what we currently have?
2. Are you suggesting getting one list of all users that are approvers/reviewers? Not sure what the benefit for it will be, eventually only an approver can `/approve` so better choose an approver to begging with no?!

Reply:
> 1. Why is it a "faster lookup" than what we currently have?

Looking up a key in a hashmap is O(1), while going through the list is O(n).

> 2. Are you suggesting getting one list of all users that are approvers/reviewers? Not sure what the benefit for it will be, eventually only an approver can /approve so better choose an approver to begging with no?!

I am suggesting getting one set, not one list.
The thinking behind merging reviewers and approvers is that you want to get someone with some kind of relationship in the project. I don't think it has to be an approver - the assignee can work on the issue / PR, then ping someone else to `/approve`.
We can stick to `approvers` only if you disagree, but I would anyway build a set of usernames for faster lookups.

Reply:
I didn't understand that you mean to use a map. I don't mind to update it and to use reviewers as well. Will be next week though.

Reply:
How will you pick a random entry in a map? For a slice it is easy...

Reply:
Just after creating the map, we could get all keys into a slice and keep that for that use case.

Reply:
As agreed offline. We will only keep the slices as they exist in the OWNER file and iterate over them ATM.
If in the future, it becomes a bottle neck we will add some additional code to make it work faster.